src/mobile/ios/Argon2/Argon2IOS.swift

```diff
@@ -18,18 +18,28 @@ class Argon2IOS: NSObject {
   ///   - password: Password to hash
   ///   - resolve: A JS Promise resolve block
   ///   - reject: A JS Promise reject block
-  @objc func hash(_ password: String, salt: String, params: [String: Any], resolver resolve: RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) {
+  @objc func hash(_ password: NSArray, salt: String, params: [String: Any], resolver resolve: RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) {
     // Validate parameters
+
     if params["t_cost"] is Int && params["m_cost"] is Int && params["parallelism"] is Int && params["hashLength"] is Int {
       // Resolve the hash
-      let h = Argon2Core.argon2Hash(password: password, salt: salt, params: params)
-      resolve(h)
+      let passwordLength = password.count as Int
```

Comment:
This cast isn't necessary
```suggestion
      let passwordLength = password.count
```